pkg/daemon/daemon.go

```diff
@@ -451,18 +485,42 @@ func (dn *Daemon) applyNodePTPProfiles() error {
 	})
 
 	relations := reconcileRelatedProfiles(dn.ptpUpdate.NodeProfiles)
+	// TODO: resolve clock IDs, clockType, leadingInterface and upstreamPort from hardware config
+	// (needed to keep code compatibility elsewhere and allow it to work both with hardware config and plugins)
 	for _, profile := range dn.ptpUpdate.NodeProfiles {
+		glog.Infof("Processing profile: %s", *profile.Name)
+
+		// Log profile details for debugging
+		if profile.Interface != nil {
+			glog.Infof("Profile %s interface: %s", *profile.Name, *profile.Interface)
+		} else {
+			glog.Infof("Profile %s has no interface field (nil)", *profile.Name)
+		}
+		if profile.Ptp4lOpts != nil {
+			glog.Infof("Profile %s ptp4lOpts: %s", *profile.Name, *profile.Ptp4lOpts)
+		}
+		if profile.Phc2sysOpts != nil {
+			glog.Infof("Profile %s phc2sysOpts: %s", *profile.Name, *profile.Phc2sysOpts)
+		}
+		if profile.Ts2PhcOpts != nil {
+			glog.Infof("Profile %s ts2phcOpts: %s", *profile.Name, *profile.Ts2PhcOpts)
+		}
 
 		if controlledID, ok := relations[*profile.Name]; ok {
 			profile.PtpSettings["controlledId"] = strconv.Itoa(controlledID)
 		}
+
+		glog.Infof("Calling applyNodePtpProfile for profile %s with runID %d", *profile.Name, runID)
 		err := dn.applyNodePtpProfile(runID, &profile)
 		if err != nil {
+			glog.Errorf("Failed to apply profile %s: %v", *profile.Name, err)
 			return err
 		}
+		glog.Infof("Successfully applied profile: %s", *profile.Name)
 		runID++
 	}
 
+	glog.Infof("All profiles applied, starting %d processes", len(dn.processManager.process))
```

Comment:
This does not include gpsd or gpspipe in the count, right?

Reply:
No, i think it does, because of this:
https://github.com/k8snetworkplumbingwg/linuxptp-daemon/blob/cb726a15f98c768d73e962fb5ede63ca7d5b86a0/pkg/daemon/daemon.go#L905